imkar/scattering/analytical.py

```diff
@@ -0,0 +1,179 @@
+import numpy as np
+import pyfar as pf
+
+import numbers
+
+
+def rectangular(frequencies, phis, width, length, height, c=343):
+    """
+    This function computes the scattering coefficient of periodic rectangular
+    profiles, assuming that the extent is infinite.
+
+    This function is based on Ducourneau's method described by Prof. Embrechts
+    [#]_.
+
+    Parameters
+    ----------
+    frequencies : array, double
+        Vector of frequencies the scattering coefficient is computed at.
+    phis : array, double
+        Vector of incidences phi the scattering coefficient is computed at.
+        Values between 0°-90° allowed.
+    width : double
+        Width of the rectangulars in m.
+    length : double
+        Length of the regtangulars in m.
+    height : double
+        Height of the profile structure.
+    c : double
+        Speed of sound with which the scattering coefficient is computed.
+    Returns
+    -------
+    s : FrequencyData
+        The analytical solution for the scattering coefficient of a
+        rectangular profile at the given incidences and frequencies.
+
+    References
+    ----------
+    .. [#]  J. Embrechts and A. Billon, "Theoretical Determination of the
+            Random-Incidence Scattering Coefficients of Infinite Rigid
+            Surfaces with a Periodic Rectangular Roughness Profile",
+            Acta Acustica united with Acustica, Bd. 97, Nr. 4, S. 607-617,
+            Juli 2011, doi:10.3813/AAA.918441
+
+    Comment
+    -------
+    In this solution the parameter r in equation (9) is also limited to
+    -N <= r <= N. The width of the well is described by the width parameter and
+    replaces the parameter 2A.
+
+    """
+
+    # check inputs
+    if not isinstance(frequencies, np.ndarray) or\
+            not isinstance(frequencies.item(0), numbers.Real):
+        raise TypeError("frequency_vector has to be an array of real numbers")
+    if frequencies.ndim > 1:
+        raise ValueError("frequency_vector has to be 1-dimensional")
+    if not isinstance(phis, np.ndarray) or \
+            not isinstance(phis.item(0), numbers.Real):
+        raise TypeError("phis has to be an array of real numbers")
+    if phis.ndim > 1:
+        raise ValueError("phis has to be 1-dimensional")
+    if (np.any(phis <= 0)) or (np.any(phis >= 90)):
+        raise ValueError("phis values have to be between 0° and 90°")
+    if not isinstance(width, numbers.Real) or width <= 0:
```

Comment:
```suggestion
    if not isinstance(width, float) or width <= 0:
```

Reply:
Changing numbers.real to float lets the test fail if the input is is given as an integer (e.g. if incident angle is declared as 30) even though it should be allowed. isinstance(numbers.real) allows all real number types